python/uv/private/lock.bzl

```diff
@@ -15,109 +15,242 @@
 """A simple macro to lock the requirements.
 """
 
-load("@bazel_skylib//rules:write_file.bzl", "write_file")
+load("@bazel_skylib//rules:diff_test.bzl", "diff_test")
+load("@bazel_skylib//rules:expand_template.bzl", "expand_template")
 load("//python:py_binary.bzl", "py_binary")
 load("//python/private:bzlmod_enabled.bzl", "BZLMOD_ENABLED")  # buildifier: disable=bzl-visibility
 
 visibility(["//..."])
 
-_REQUIREMENTS_TARGET_COMPATIBLE_WITH = select({
-    "@platforms//os:windows": ["@platforms//:incompatible"],
-    "//conditions:default": [],
-}) if BZLMOD_ENABLED else ["@platforms//:incompatible"]
+_REQUIREMENTS_TARGET_COMPATIBLE_WITH = [] if BZLMOD_ENABLED else ["@platforms//:incompatible"]
 
-def lock(*, name, srcs, out, upgrade = False, universal = True, args = [], **kwargs):
+_LockRunInfo = provider(
+    doc = "Information about source tree for Sphinx to build.",
+    fields = {
+        "args": "",
+        "cmd": "",
+        "cmd_file": "",
+        "srcs": "",
+    },
+)
+
+def _lock_impl(ctx):
+    args = ctx.actions.args()
+    if ctx.files.maybe_out:
+        args.add_all([
+            "--src-out",
+            ctx.files.maybe_out[0].path,
+        ])
+    args.add("--output-file", ctx.outputs.out)
+    args.add_all(ctx.files.srcs)
+    args.add_all(["--custom-compile-command", "bazel run //{}:{}.update".format(
+        ctx.label.package,
+        ctx.label.name,
+    )])
+    args.add_all([
+        "--no-python-downloads",
+        "--no-cache",
+    ])
+    args.add_all(ctx.attr.args)
```

Comment:
I think adding `uv` path here would be a good way to get rid of the need for `runfiles` if we decided to drop the current structure.